cosmoz-default-tree.html

```diff
@@ -23,134 +34,192 @@
-			return;
+	/**
+	 * Returns the node of a given path.
+	 * @return {object}
+	 * @param {string} pathLocator (The string which describes the path. e.g. "1.2.9")
+	 * @param {object} nodeObj (The object the search should be based on.) Default: this._treeData
+	 * @param {string} separatorSign (The string which separates the path. e.g ".") Default: "."
+	 * @param {string} childProperty (The string which describes the child property of a node.) Default: "children"
+	*/
+	Cosmoz.DefaultTree.prototype.getNodeByPathLocator = function (pathLocator, nodeObj, separatorSign, childProperty) {
+		if (!pathLocator) {
+			return this._treeData;
 		}
 
-		if (propertyName === 'pathLocator') {
-			return this._getPathByPathLocator(propertyValue);
-		}
+		// Defaults
+		nodeObj = nodeObj ? nodeObj : this._treeData;
+		separatorSign = separatorSign ? separatorSign : '.';
+		childProperty = childProperty ? childProperty : 'children';
 
-		var node = this._searchAllRoots(propertyName, propertyValue);
-		if (node !== undefined) {
-			return this._getPathByPathLocator(node.pathLocator);
-		}
+		var	pathNodes = this.getPathNodes(pathLocator, nodeObj, separatorSign, childProperty),
+			lastNodeOnPath = pathNodes.pop();
+		return lastNodeOnPath !== undefined ? lastNodeOnPath : null;
 	};
 
-	Cosmoz.DefaultTree.prototype.getChildren = function (node) {
-		if (node === undefined) {
-			return;
+	/**
+	 * Returns the nodes on a given path.
+	 * @return {array} or null if there was an empty response.
+	 * @param {string} pathLocator (The string which describes the path. e.g. "1.2.9")
+	 * @param {object} nodeObj (The object the search should be based on.) Default: this._treeData
+	 * @param {string} separatorSign (The string which separates the path. e.g ".") Default: "."
+	 * @param {string} childProperty (The string which describes the child property of a node.) Default: "children"
+	*/
+	Cosmoz.DefaultTree.prototype.getPathNodes = function (pathLocator, nodeObj, separatorSign, childProperty) {
+		if (!pathLocator) {
+			return this._roots;
 		}
-		return _objectValues(node.children);
-	};
 
-	Cosmoz.DefaultTree.prototype.getProperty = function (node, propertyName) {
-		if (node === undefined || propertyName === undefined) {
-			return;
-		}
+		// Defaults
+		separatorSign = separatorSign ? separatorSign : '.';
+		childProperty = childProperty ? childProperty : 'children';
+
+		var path = pathLocator.split(separatorSign),
+			pathSegment = nodeObj ? nodeObj : this._treeData,
+			pathIsInvalid,
+			nodes;
+		
+		// Get the nodes on the path.
+		nodes = path.map(function (nodeKey) {
+			var node = pathSegment[nodeKey],
+				children = node[childProperty];
+			if (node && children !== undefined && Object.keys(children).length > 0) {
+				pathSegment = children;
+			}
+			return node;
+		}, this);
 
-		return node[propertyName];
-	};
+		// Check if all nodes on the path are valid.
+		pathIsInvalid = nodes.some(function (n) {
```

Comment:
The logic doesn't start traversing a path until without finding a hit, so you wouldn't ever be able to just "skip" node `21` and find the path beneath it. (Also this is not a relevant scenario in our app.)
But, thinking about it, using a `1.9.20.5.7` path will put `undefined` in the middle which is horrible..
And even though the current implementation of `getNodeByPathLocator()` will get `undefined` at the end of the array for a bad path, causing `getNodeByPathLocator()` to return `null`, we should probably have a clearer error handling of this.

But still, any `undefined` in the beginning of a path is probably ok, so we might want to filter those out before doing this check:
```js
while (nodes.length > 0 && nodes[0] === undefined) {
  nodes.shift();
}
```

Unless we have a reason to keep them, then we might want to make the path validator function more intelligent instead.

Also, to avoid logic like "`!pathIsInvalid`" which hurts readability (and the mind) we should probably invert this to something like
```js
pathIsValid = nodes.every(function (node) {
  return node !== undefined;
}
```
(This also matches your code comment closer.)